filebeat/processor/add_kubernetes_metadata/matchers_test.go

```diff
@@ -125,6 +125,22 @@ func TestLogsPathMatcher_InvalidVarLogPodSource(t *testing.T) {
 	executeTestWithResourceType(t, cfgLogsPath, cfgResourceType, source, expectedResult)
 }
 
+func TestLogsPathMatcher_ValidVarLogPodSource(t *testing.T) {
+	cfgLogsPath := "/var/log/pods/"
+	cfgResourceType := "pod"
+	source := fmt.Sprintf("/var/log/pods/namespace_pod-name_%s/container/0.log.20220221-210912", puid)
+	expectedResult := puid
+	executeTestWithResourceType(t, cfgLogsPath, cfgResourceType, source, expectedResult)
+}
+
+func TestLogsPathMatcher_InvalidVarLogPodSource2(t *testing.T) {
+	cfgLogsPath := "/var/log/pods/"
+	cfgResourceType := "pod"
+	source := fmt.Sprintf("/var/log/pods/namespace_pod-name_%s/container/0.log.20220221-210526.gz", puid)
+	expectedResult := ""
```

Comment:
I understood from the description that we want to  get the pod uid from the rotated log files

Reply:
Oh. Ok we don't want the .gz files though

Reply:
for the `resource_type: container` log files are explicitly checked: `strings.HasSuffix(source, ".log")` (as for each container will be only 1 file - the symlink to the log in /var/log/pods/*/*.log`), but for pod logs - there might be `*.gz` files:
```
/var/log/pods/kube-system_metricbeat-56nf2_64e94e05-c377-4bcb-9cc2-5184c4f4b978/metricbeat# ls -l | awk -F " " '{print $9}'

0.log
0.log.20220221-203122.gz
0.log.20220221-203252.gz
0.log.20220221-210526.gz
0.log.20220221-210912
1.log
1.log.20220308-190811.gz
1.log.20220308-190916.gz
1.log.20220308-191043.gz
1.log.20220308-191159
```
and it doesn't make sense to process them